### PR TITLE
go/runtime: Support bundle components without ELF binary

### LINKED ISSUE #5780: Support bundle components without ELF binary
Currently the assumption is that each bundle component always includes an ELF binary. This should not be required (e.g. it doesn't make sense for ROFL apps) and it may also not be always possible in the future (e.g. for Intel TDX). It should also make the runtime provisioner default to the TEE environment to simplify configuration.

### PR DESCRIPTION
**Pull request by kostko**
Fixes #5780

### CODE REVIEW

File: go/runtime/registry/config.go [hunk 1]
```diff
@@ -102,7 +102,62 @@ func newConfig( //nolint: gocyclo
...
+			switch bnd.Manifest.IsDetached() {
+			case false:
+				// A regular non-detached bundle that has the RONL component.
+				regularBundles = append(regularBundles, bnd)
+			case true:
+				// A detached bundle without the RONL component that needs to be attached.
+				detachedBundles[bnd.Manifest.ID] = append(detachedBundles[bnd.Manifest.ID], bnd)
+
+				// Ensure there are no name conflicts among the components.
+				for compID := range bnd.Manifest.GetAvailableComponents() {
```
Comment (peternose) [inline go/runtime/registry/config.go:140]: I forgot why was this not moved outside of the `switch`, to also check that we have only one RONL component for runtime and to also take into account components from non-detached bundles? :thinking:

**Review by peternose — APPROVED** [review]

File: go/runtime/registry/config.go (hunk 1 above)
Comment (kostko) [reply go/runtime/registry/config.go:140]: Inside the same bundle this is already checked, so this is only needed when you have detached bundles. Unless I'm missing something.

Comment (kostko) [reply go/runtime/registry/config.go:140]: > Can a non-detached bundle (the one which contains RONL) also contain components

Yes.

> and do we need to compare them against detached for duplicates

Components in the RONL bundle currently override any components from detached bundles, see below:
```rust
// Skip components that already exist in the bundle itself.
if bnd.Manifest.GetComponentByID(detachedComp.ID()) != nil {
	continue
}
```

> Could you have 2 non-detached bundles with the same RONL, do we capture this?

Yes, you can have multiple bundles but they need to be for different versions. If they are for the same version this is an error captured below:
```rust
if _, ok := rh.Runtimes[id][bnd.Manifest.Version]; ok {
	return nil, fmt.Errorf("duplicate runtime '%s' version '%s'", id, bnd.Manifest.Version)
}
```

Comment (peternose) [reply go/runtime/registry/config.go:140]: OK, than everything works as expected.

**Pull request merged**